src/kernel/mempool_limits.h

```diff
@@ -17,13 +17,21 @@ namespace kernel {
  */
 struct MemPoolLimits {
     //! The maximum allowed number of transactions in a package including the entry and its ancestors.
-    int64_t ancestor_count{DEFAULT_ANCESTOR_LIMIT};
+    uint64_t ancestor_count{DEFAULT_ANCESTOR_LIMIT};
     //! The maximum allowed size in virtual bytes of an entry and its ancestors within a package.
-    int64_t ancestor_size_vbytes{DEFAULT_ANCESTOR_SIZE_LIMIT_KVB * 1'000};
+    uint64_t ancestor_size_vbytes{DEFAULT_ANCESTOR_SIZE_LIMIT_KVB * 1'000};
     //! The maximum allowed number of transactions in a package including the entry and its descendants.
-    int64_t descendant_count{DEFAULT_DESCENDANT_LIMIT};
+    uint64_t descendant_count{DEFAULT_DESCENDANT_LIMIT};
     //! The maximum allowed size in virtual bytes of an entry and its descendants within a package.
-    int64_t descendant_size_vbytes{DEFAULT_DESCENDANT_SIZE_LIMIT_KVB * 1'000};
+    uint64_t descendant_size_vbytes{DEFAULT_DESCENDANT_SIZE_LIMIT_KVB * 1'000};
+
+    /**
+     * @return MemPoolLimits with all the limits set to the maximum
+     */
+    static MemPoolLimits NoLimits() {
+        uint64_t no_limit{std::numeric_limits<uint64_t>::max()};
+        return {no_limit, no_limit, no_limit, no_limit};
+    }
```

Comment:
I think it would make sense to delete the default ctor and require callers to set all 4 members when constructing.

Reply:
What would be the benefit of that approach? Asking because it's quite a deviation from the current process, where we construct an empty MemPoolLimits (with default values as defined here) in init.cpp (as a member of CTxMemPool::Options), and then pass it to multiple AppluArgsManOptions() functions to update it with startup parameters if provided. I think deleting the default ctor of MemPoolLimits would require changing that whole flow, and I'm not sure it's worth it. I think having the default values here and not requiring every value to be explicitly defined makes sense?